templates/nginx-ingress.go

```diff
@@ -203,6 +203,7 @@ spec:
             - --tcp-services-configmap=$(POD_NAMESPACE)/tcp-services
             - --udp-services-configmap=$(POD_NAMESPACE)/udp-services
             - --annotations-prefix=nginx.ingress.kubernetes.io
+            - --enable-ssl-passthrough
```

Comment:
Since the default is to disable it, should we just disable the ingress controller in the yaml and deploy it as an addon in the pass through?

Reply:
Could be, but not sure, due to ssl-passthrough is not mandatory. It just give the option to use annotation `nginx.ingress.kubernetes.io/ssl-passthrough` at ingress definition, `false` by default.

Reply:
@rawmind0 I think adding arguments is going to a be common problem with the nginx controller.  Today we have options as below to control nginx options, but it does not cover arguments.  How about we add `ExtraArguments` where you can add in `--enable-ssl-passthrough`

```yaml
ingress:
  provider: nginx
  options:
  - foo: bar
  extra_arguments:
  - --enable-ssl-passthrough
```

Reply:
@ibuildthecloud , `ExtraArguments` better option.
I'll update this PR once https://github.com/rancher/types/pull/366 is merged.